src/extension/app/components/plugin/env-switcher/env-switcher.js

```diff
@@ -222,56 +261,62 @@ export class EnvironmentSwitcher extends ConnectedElement {
       showProd = true;
     }
 
+    if (showNotification && this.currentEnv !== 'edit') {
```

Comment:
No I think you are right, good call. 

I'll keep the notification but hide the publish button if it doesn't meet the condition. Wdyt?

Reply:
Ended up disabling the button instead like it is in the action bar. Lmk  if that doesn't work for you.

Reply:
This might create confusion with customers who hide the publish button under certain circumstances (e.g. when the document is in `drafts`. I think we have to check both `condition` and `isEnabled` so we can react properly.